### ISSUE TITLE
Custom easing function error

### ISSUE THREAD

**Issue by nestorrente** [koggdal/ocanvas#111]
When I define a custom easing function, oCanvas breaks. The error is "Uncaught TypeError: Cannot read property 'length' of undefined".

I think that the error comes from the function "parseEasingOption". The code of the function is:

parseEasingOption: function (easing) {
	if (typeof easing === "string") {
		// The cubic-bezier() syntax is now deprecated (though it was never really public)
		if (~easing.indexOf("cubic-bezier")) {
			return this.getCustomCubicBezier(easing) || this.easing[this.defaults.easing];
		} else {
			return this.easing[easing] || this.easing[this.defaults.easing];
		}
	} else if (typeof easing !== "function") {
		return this.easing[this.defaults.easing];
	}
},

I think that the error is because an "else" statement is missing. Adding an "else return easing;" solves the problem.

**Comment by Canopee** [comment]
Bonjour,

Actuellement absent du bureau , je prendrai connaissance de votre message à mon retour le lundi 17 août 2015.

Cordialement,
Esteves Alexis
Développeur numérique

**Comment by nestorrente** [comment]
Ok, I was confused. Now I know I cannot define a easing as a pure JavaScript function. I must define it as a string like "cubic-bezier(0.665, 0.005, 0.265, 1)".

However, I think that would be good adding this custom-javascript-function feature.

**Comment by nestorrente** [comment]
Well, I was confused again. The syntax "cubic-bezier(...)" is now deprecated.

I solved my problem using one of the default easing functions, but I would like to be able to define my own easing function.

Thanks for developing oCanvas, I love it!

**Comment by koggdal** [comment]
So sorry about not responding earlier!

I must have completely forgotten about custom easing functions when the syntax was changed for animations. The docs still specify it should work in the new syntax, and I really think it should. So I added support for it now. :) It's on master. I'll have a look at the PR about clipping too and then decide if this should be in its own release or wait for the clipping too.

**Comment by koggdal** [comment]
It's now released as 2.7.5. From [the updated docs](http://ocanvas.org/docs/Animation/animate#easing-custom-function):

> If a function is used, it will be called with four arguments: current time (in milliseconds), start value, change in value (from start to end) and total duration of the animation (in milliseconds). It should return the value to use for the current position.

So you can now use it like this (for a linear easing function):

```js
object.animate({rotation: 180}, {
  duration: 1000,
  easing: function(time, start, change, duration) {
    return time / duration * change;
  }
});
```

It will soon appear on the CDN.

**Comment by koggdal** [comment]
https://cdnjs.cloudflare.com/ajax/libs/ocanvas/2.7.5/ocanvas.min.js
